mlir/lib/Dialect/AMDGPU/IR/AMDGPUDialect.cpp

```diff
@@ -43,6 +43,21 @@ void AMDGPUDialect::initialize() {
       >();
 }
 
+bool amdgpu::AMDGPUDialect::isSharedMemoryAddressSpace(Attribute memorySpace) {
+  if (!memorySpace)
+    return false;
+  if (auto intAttr = llvm::dyn_cast<IntegerAttr>(memorySpace))
+    return intAttr.getInt() == AMDGPUDialect::kSharedMemoryAddressSpace;
```

Comment:
Other than at the LLVM level, I wouldn't think we'll have integer based memory spaces.

Reply:
Yeah, but roundtripping/backcompat/...

Reply:
Thanks for the comments above. Krzysztof, that make sense. For backward compatibility reasons, can we leave like this?

Reply:
Please let me know when you can get back on this.

Reply:
I'm fine with having the check for integral address spaces, if that helps

Reply:
Thanks.